clippy_lints/src/methods/manual_contains.rs

```diff
@@ -0,0 +1,117 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::peel_hir_pat_refs;
+use clippy_utils::source::snippet_with_applicability;
+use rustc_ast::UnOp;
+use rustc_errors::Applicability;
+use rustc_hir::def::Res;
+use rustc_hir::{BinOpKind, Body, Expr, ExprKind, HirId, QPath};
+use rustc_lint::LateContext;
+use rustc_middle::ty::{self};
+use rustc_span::Span;
+use rustc_span::source_map::Spanned;
+
+use super::MANUAL_CONTAINS;
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, recv: &Expr<'_>, closure_arg: &Expr<'_>) {
+    let mut app = Applicability::MachineApplicable;
+
+    if !expr.span.from_expansion()
+        // check if `iter().any()` can be replaced with `contains()`
+        && let ExprKind::Closure(closure) = closure_arg.kind
+        && let Body{params: [param],value} = cx.tcx.hir().body(closure.body)
+        && let ExprKind::Binary(op, lhs, rhs) = value.kind
+        && let (peeled_ref_pat, _)=peel_hir_pat_refs(param.pat)
+        && let Some((snip,snip_expr))=can_replace_with_contains(cx, op, lhs, rhs, peeled_ref_pat.hir_id, &mut app)
+    {
+        let ref_type = cx.typeck_results().expr_ty_adjusted(recv);
+
+        match ref_type.kind() {
+            ty::Ref(_, inner_type, _) if inner_type.is_slice() => {
+                if let ty::Slice(slice_type) = inner_type.kind()
+                    && *slice_type == cx.typeck_results().expr_ty(snip_expr)
+                {
+                    span_lint_and_sugg(
+                        cx,
+                        MANUAL_CONTAINS,
+                        expr.span,
+                        "using `contains()` instead of `iter().any()` is more efficient",
+                        "try",
+                        format!(
+                            "{}.contains({})",
+                            snippet_with_applicability(cx, recv.span, "..", &mut app),
```

Comment:
Placeholder for a single expression is better expressed with `"_"`:

```suggestion
                            snippet_with_applicability(cx, recv.span, "_", &mut app),
```